app/lib/bk/compat/parsers/bitrise/steps.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+module BK
+  module Compat
+    module BitriseSteps
+      # Implementation of Bitrise step translations
+      class Translator
+        VALID_STEP_TYPES = %w[change-workdir git-clone script].freeze
+
+        def matcher(type, _inputs)
+          VALID_STEP_TYPES.include?(type.downcase)
+        end
+
+        def translator(type, inputs)
+          send("translate_#{type.downcase.gsub('-', '_')}", inputs)
+        end
+
+        def translate_change_workdir(inputs)
+          generate_change_workdir_command(inputs)
+        end
+
+        def generate_change_workdir_command(inputs)
```

Comment:
why have a method that does nothing and just calls another one?

```suggestion
        def translate_change_workdir(inputs)
```

Reply:
Looking at the code, I think that `generate_<step type>_command` might not actually work the best in the long run, it was more to split the underlying generation to defs to relay (and in the case of #151 thats split into a `commands.rb` just to make `steps.rb` not do SLOC ballooning)

I've reverted that decision in [cf59c3a](https://github.com/buildkite/migration/pull/141/commits/cf59c3a954c1381b4220f9b16976e0fb60c9efc0), will need to assess #151 based on the decision we take here